docs/guides/isolated_systems.md

```diff
@@ -0,0 +1,138 @@
+# Isolated systems guide
+
+SQLMesh is optimized for use in systems where developers have access to production data.
+
+Writing code against partial or unrepresentative data can cause problems because you don't become aware of changes in production data until errors have already occurred. Other data products, such as machine learning models, may depend on the distribution of values in the training data - building them on unrepresentative data may lead to different behavior in production than in development.
+
+However, some companies store production and non-production data in different data warehouses that can't talk to one another ("isolated systems"). This is usually due to information security concerns, as the non-production warehouse may be accessible to more users and/or have looser security restrictions.
+
+This guide explains how to use SQLMesh with isolated systems and how isolating systems affects SQLMesh's behavior.
+
+## Terminology
+
+Isolated systems are sometimes referred to as "isolated environments," but we avoid that term because "environments" has a specific meaning in SQLMesh.
+
+Instead, we will refer to them as isolated systems - the "production system" and "non-production system."
+
+When we refer to "environments," we are always talking about [SQLMesh environments](../concepts/environments.md) - the isolated namespaces created and managed by SQLMesh.
+
+## Configuring SQLMesh
+
+### Separate state data
+
+SQLMesh maintains a record of every model version so it can identify changes when models are updated. Those records are called "state" data, as in "the state of the model at that point in time."
+
+State data can be stored alongside other data in the primary data warehouse or in a [separate database](./connections.md#state-connection). (We recommend using a separate transactional database for projects running on cloud SQL engines.)
```

Comment:
Would this be better to link to? https://sqlmesh.readthedocs.io/en/stable/guides/configuration/#state-connection

Or maybe some of that information should be copied over?